### ISSUE TITLE
[a11y]: Keyboard navigation in ProductiveCard is not working as expected

### ISSUE THREAD

**Issue by mariyageorge01** [carbon-design-system/ibm-products#7705]
### Package

@carbon/ibm-products

### Browser

Chrome

### Operating System

MacOS

### Package version

2.64.0

### Automated testing tool and ruleset

Keyboard navigation

### Assistive technology

None used

### Description

I have Productive cards and clicking on the Productive card's overflow menu -> overflow menu item should take me to a Modal with a TextInput. The issue I encountered is that while opening the overflow menu in the Productive card via keyboard, the focus doesn't go to the Modal's TextInput. 

Note: If you open the overflow menu by mouse click, there are no focus issues.

### WCAG 2.1 Violation

2.1.1 Keyboard

### Reproduction/example

https://stackblitz.com/edit/github-alee8pr9?file=src%2FExample%2FExample.jsx%3AL57,package.json

### Steps to reproduce

1. Open ProductiveCard's overflow menu via keyboard.
2. Click on ProductiveCard's overflow menu item via keyboard

Expected behaviour: Focus should be on modal's TextInput

Observed behaviour: Focus goes to the second Productive card's overflow menu

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems

**Comment by davidmenendez** [comment]
hey @mariyageorge01 thanks for opening up this issue. so this is actually an implementation bug. `onKeyDown: action` is the culprit here. if you remove it from the actions it works as intended.

it looks like the `OverflowMenu` component already catches common key inputs like return or space to activate the `onClick` method via keyboard. `onKeyDown` appears to supersede it and is probably causing whatever issue with the focus mechanism. so i would just remove that and you should be good to go!

when you get a chance can you please verify this so we can close this issue? thanks!

**Comment by davidmenendez** [comment]
since we haven't heard back and this appears to just be an user implementation issue i'm going to mark this as closed.
